gwctl/pkg/printer/httproutes.go

```diff
@@ -48,7 +51,17 @@ func (hp *HTTPRoutesPrinter) Print(resourceModel *resourcediscovery.ResourceMode
 			hostNamesOutput = fmt.Sprintf("%v + %v more", strings.Join(hostNames[:2], ","), cnt-2)
 		}
 
-		row := []string{httpRouteNode.HTTPRoute.Name, hostNamesOutput}
+		parentRefsCount := fmt.Sprintf("%d", len(httpRouteNode.HTTPRoute.Spec.ParentRefs))
+
+		age := duration.HumanDuration(time.Since(httpRouteNode.HTTPRoute.GetCreationTimestamp().Time))
```

Comment:
Seeing our growing dependence on outputs affected by the "current time", maybe it would be a good time to provide a way to inject a fake clock during testing? ... something like an additional `clock` field in the `HTTPRoutesPrinter` struct? (Similar thoughts on the other PRs)

There are some standard interfaces available at https://github.com/kubernetes/utils/blob/master/clock/clock.go along with some fake implementations at https://github.com/kubernetes/utils/tree/master/clock/testing.

Reply:
@gauravkghildiyal Absolutely! I completely agree that providing a way to inject a fake clock during testing would be incredibly valuable. It would allow us to isolate specific functionalities and test them independently of the actual time, improving the reliability and efficiency of our tests. I'd love to learn more about implementing this and contribute to the solution. It would be a great learning experience.